docs/technical-overview.md

```diff
@@ -0,0 +1,267 @@
+
+All model classes (like `Fixture`, `Manufacturer`, `Physical`, `Channel` or `Range`) are located in the `lib/model/` directory. When using the model, it should be enough importing the `Fixture` module; instances of other classes are returned by the fixture's properties:
+
+```js
+const Fixture = require('lib/model/Fixture.js');
+
+const myFix = Fixture.fromRepository('cameo', 'nanospot-120'); // instanceof Fixture
+
+const physicalData = myFix.physical; // instanceof Physical
+const panFine = myFix.getChannelByKey('Pan fine'); // instanceof FineChannel
+
+if (panFine.coarseChannel.hasHighlightValue) {
+  console.log(`Hightlight at ${panFine.coarseChannel.highlightValue}`)
+}
+```
+
+Model properties are always implemented using getters and setters. To store data, we use backing fields (prefixed with underscore, like `_jsonObject`) that should never be invoked from outside. Properties that need further computation or create other objects should be cached in the internal `_cache` object. Avoid returning `undefined` under all circumstances! To achieve this, use defaults for optional properties – if the default is not `null`, it's good practice providing a `hasXY` boolean.
+
+*It is planned* to document all available classes with their properties and methods using [JSDoc](http://usejsdoc.org/). That's why JSDoc annotations are required for new properties/methods and should be gradually added to existing ones.
+
+```js
+module.exports = class Fixture {
+  set jsonObject(jsonObject) {
+    this._jsonObject = jsonObject;
+    this._cache = {};
+  }
+
+  // only returns backing field to avoid accessing _jsonObject from outside
+  get jsonObject() {
+    return this._jsonObject;
+  }
+
+  // required, no default needed
+  get name() {
+    return this._jsonObject.name;
+  }
+
+  // defaults to the name
+  get shortName() {
+    return this._jsonObject.shortName || this._jsonObject.name;
+  }
+
+  get hasShortName() {
+    return 'shortName' in this._jsonObject;
+  }
+
+  // it's not good creating a Meta object with each property access, so we cache it
+  get meta() {
+    if (!('meta' in this._cache)) {
+      this._cache.meta = new Meta(this._jsonObject.meta);
+    }
+
+    return this._cache.meta;
+  }
+
+  // defaults to null as there is no meaningful other default
+  get manualURL() {
+    return this._jsonObject.manualURL || null;
+  }
+
+  // ...
+}
+```
+
+## Fixture features
+*[⬆️ Back to top](#technical-overview)*
+
+Fixture features are specific [fixture](#fixtures) characteristics (like "uses RDM" or "uses fine channel before coarse channel"), especially ones that have produced or are likely to produce bugs and errors.
+
+We use fixture features for the following purposes:
+- Generate a possibly short variety of test fixtures that use possibly many different fixture features so we can check all special cases
```

Comment:
Generate a minimal collection of test fixtures that cover all fixture features so we can check all special cases with minimum work